test/ui/cypress/code/environment.js

```diff
@@ -20,6 +22,64 @@ class Environment {
   namespace() {
     return Cypress.env('TEST_NAMESPACE')
   }
+
+  random() {
+    if (Environment.#rand === undefined) {
+      let seed = Cypress.env('TEST_RANDOM')
+      if (seed === undefined) {
+        seed = (Math.random() + 1).toString(36).substring(7)
+      }
+      cy.log(`Using seed: ${seed} in testing.`)
+      cy.log(`Set TEST_RANDOM=${seed} to recreate this test.`)
+      Environment.#rand = new PRNG(seed)
+    }
+    return Environment.#rand
+  }
+}
+
+class PRNG {
+  #rand
+
+  constructor(seedString) {
+    // Create xmur3 state:
+    const seed = xmur3(seedString);
+    // Output four 32-bit hashes to provide the seed for sfc32.
+    this.#rand = sfc32(seed(), seed(), seed(), seed())
+  }
+
+  next() {
+    return this.#rand()
+  }
+}
+
+// See: https://stackoverflow.com/a/47593316/844449
+function xmur3(str) {
+  for (var i = 0, h = 1779033703 ^ str.length; i < str.length; i++) {
```

Comment:
Could it be simplified as:
```
let h = 1779033703 ^ str.length;
let i;
for (; i < str.length; i++) {
...
``` 

?

Reply:
I didn't change that copied code. But, I'll be removing it after @mgencur comments.